src/VALI-XCS.cpp

```diff
@@ -105,3 +105,26 @@ try {
   PrintException(e);
   return EXIT_FAILURE;
 }
+
+#if defined __ANDROID_API__ && __ANDROID_API__ < 17
+extern "C" {
+
+  /** \brief Mimic posix_memalign as a thin wrapper around memalign.
+   *
+   * posix_memalign is not present in the API version 16 bionic library and before, but is required by the builtin "new" operator of clang in NDK19 or later.
+   */
+  int posix_memalign(void **memptr, size_t alignment, size_t size) {
```

Comment:
No duplicate code!

Reply:
Looking at it again, absolutely agree. Just being lazy.
Will remove that commit, and instead add the source to the program.